libminifi/include/io/validation.h

```diff
@@ -27,6 +27,21 @@
  * A checker that will, at compile time, tell us
  * if the declared type has a size method.
  */
+template<typename T>
+class empty_function_functor_checker {
```

Comment:
What's the reason for this? To use empty(), which is guaranteed to be O(1), while size() doesn't?

Reply:
yep just a switch to empty.